apps/agent-tars/src/renderer/src/agent/Executor/index.ts

```diff
@@ -87,16 +87,27 @@ You should use the same language as the user input by default.
         resolve([]);
       };
 
+      const activeMcpSettings = await ipcClient
+        .getActiveMcpSettings()
+        .catch((e) => {
+          console.error('Error getting active MCP settings', e);
+          return {};
+        });
       try {
         this.abortSignal.addEventListener('abort', abortHandler);
+
         const result = await ipcClient.askLLMTool({
           messages: [
             Message.systemMessage(this.systemPrompt),
             Message.userMessage(environmentInfo),
             Message.userMessage(`Aware status: ${status}`),
           ],
           tools: [idleTool, chatMessageTool],
-          mcpServerKeys: Object.values(MCPServerName),
+          mcpServerKeys: [
+            ...Object.values(MCPServerName),
+            // user defined mcp servers
+            ...Object.keys(activeMcpSettings),
```

Comment:
We can use extensions to adjust the order here and the actual mcp tools that take effect later // cc @helio9cn